sycl/source/detail/global_handler.cpp

```diff
@@ -399,6 +404,29 @@ extern "C" __SYCL_EXPORT BOOL WINAPI DllMain(HINSTANCE hinstDLL,
   }
   return TRUE; // Successful DLL_PROCESS_ATTACH.
 }
+BOOL isLinkedStatically() {
+  // If the exePath is the same as the dllPath,
+  // or if the module handle for DllMain is not retrievable,
+  // then we are linked statically
+  // Otherwise we are dynamically linked or loaded.
+  HMODULE hModule = nullptr;
+  auto LpModuleAddr = reinterpret_cast<LPCSTR>(&DllMain);
+  if (GetModuleHandleExA(GET_MODULE_HANDLE_EX_FLAG_FROM_ADDRESS, LpModuleAddr,
+                         &hModule)) {
+    char dllPath[MAX_PATH];
+    if (GetModuleFileNameA(hModule, dllPath, MAX_PATH)) {
+      char exePath[MAX_PATH];
+      if (GetModuleFileNameA(NULL, exePath, MAX_PATH)) {
+        if (std::string(dllPath) == std::string(exePath)) {
+          return true;
+        }
+      }
+    }
+  } else {
+    return true;
+  }
+  return false;
```

Comment:
This line looks unreachable

Reply:
It is reachable. Those three nested 'if' in the main consequent don't have else. They'll end up here and (correctly) return false.

Reply:
Ah, right, don't know how I managed to misread that. May be it makes sense to reduce the nesting by flipping the condition and using early returns (at least for the outermost if).

Reply:
Flipping it might make it read a bit better, and comments will help as well.